1.2/sd-client/src/main/java/ServiceDirectoryClient.java

```diff
@@ -0,0 +1,256 @@
+                        DirectoryServiceRestfulClient.SD_API_SD_SERVER_PORT_PROPERTY,
+                        port);
+            }
+            if (line.hasOption("exec")) {
+                String commandString = line.getOptionValue("exec");
+                try {
+                    cmdArgs = parseCommand(commandString);
+                    cmd = CMD.valueOf(cmdArgs[0]);
+                    if (cmd.sizeOfArgs + 1 != cmdArgs.length) {
+                        throw new ParseException(
+                                String.format("Wrong arguments number for command %s, required %s but %s \n",
+                                        cmdArgs[0], cmd.sizeOfArgs, cmdArgs.length - 1));
+                    }
+                    switch (cmd) {
+                        case getInstanceOf:
+                            lookupInstance(cmdArgs);
+                            break;
+                        case getAllInstancesOf:
+                            lookupInstances(cmdArgs);
+                            break;
+                        case testConnection:
+                            connectToServer();
+                            break;
+                        default:
+                            fail("command " + cmd + " is unsupported now");
+                            break;
+                    }
+                } catch (IllegalArgumentException e) {
+                    throw new ParseException(String.format("Unknown command [ %s ] \n", commandString));
+                }
+            }
+        } catch (ParseException exp) {
+            // oops, something went wrong
+            System.out.println(exp.getMessage());
+            printUsage();
+        }
+    }
+
+    private static String[] parseCommand(String cmdString) {
+        StringTokenizer cmdTokens = new StringTokenizer(cmdString, " ");
+        String[] args = new String[cmdTokens.countTokens()];
+        int tokenIndex = 0;
+        while (cmdTokens.hasMoreTokens()) {
+            args[tokenIndex] = cmdTokens.nextToken();
+            tokenIndex++;
+        }
+        return args;
+    }
+
+    private static void printUsage() {
+        HelpFormatter formatter = new HelpFormatter();
+        formatter.setOptionComparator(null);
+        formatter.printHelp("SDClient", options);
+        for (Map.Entry<CMD, String> cmd : commands.entrySet()) {
+            System.out.println("\t" + cmd.getKey() + " \t" + cmd.getValue());
+        }
+    }
+
+    /**
+     * Do the connect command.
+     */
+    private void connectToServer() {
+        try {
+            new DirectoryServiceRestfulClient().getAllInstances();
+        } catch (ServiceException e) {
+            fail(e.getServiceDirectoryError().getErrorMessage());
+        }
+    }
+
+    /**
+     * Do the lookupInstance command.
+     *
+     * @return true if complete.
+     */
+    private void lookupInstance(String args[]) {
+
+        String serviceName = args[1].trim();
```

Comment:
the code need to do refactoring since it's copy it from the old sd-tool, It used up my day-time yesterday try to reuse the old parser code, and finally give up and rewrite all by using common-cli.